### PR TITLE
Pull Request of Logistic Regression Model for Diabetes Prediction

### PR DESCRIPTION
**Pull request by keshripritesh**
### 🔹 Summary

This PR adds a **Logistic Regression-based Machine Learning model** to predict the presence of **Diabetes or Pre-Diabetes** using the **Diabetes Health Indicators Dataset (BRFSS 2015)**.
The model leverages a **50-50 balanced dataset** (approximately 7,000 randomly scaled instances from the original \~70,000) to enhance performance and computational feasibility.

### 🔹 Key Features

* ✅ Logistic Regression model trained to classify **Diabetes (1) vs No Diabetes (0)**
* ✅ Dataset preprocessing and feature selection for optimized performance
* ✅ Model training and parameter storage for later use
* ✅ Predictive functionality with **retrieval of trained model parameters**
* ✅ Interactive interface built using **Streamlit**

### 🔹 Dataset

* **Balanced 50-50 split dataset (used in training)**
  🔗 [[Diabetes Binary 50-50 Split](https://www.kaggle.com/datasets/alexteboul/diabetes-health-indicators-dataset?select=diabetes_binary_5050split_health_indicators_BRFSS2015.csv)](https://www.kaggle.com/datasets/alexteboul/diabetes-health-indicators-dataset?select=diabetes_binary_5050split_health_indicators_BRFSS2015.csv)

* **Original dataset (for reference)**
  🔗 [[Diabetes 012 Dataset](https://www.kaggle.com/datasets/alexteboul/diabetes-health-indicators-dataset?select=diabetes_012_health_indicators_BRFSS2015.csv)](https://www.kaggle.com/datasets/alexteboul/diabetes-health-indicators-dataset?select=diabetes_012_health_indicators_BRFSS2015.csv)

### 🔹 Features Used

* Health-related: `HighBP`, `HighChol`, `CholCheck`, `BMI`, `Smoker`, `Stroke`, `HeartDiseaseOrAttack`, `PhysActivity`, `Fruits`, `Veggies`, `HvyAlcoholConsmp`, `AnyHealthCare`, `NoDocbcCost`, `GenHlth`, `MentHealth`, `PhysHealth`, `DiffWalk`
* Target variable: `Diabetes_binary`

### 🔹 Tech Stack

* **Languages/Tools**: Python, Jupyter Notebook
* **Libraries**: Numpy, Pandas, Matplotlib, Sklearn, Streamlit

### 🔹 How to Run

1. Clone the repository

   ```bash
   git clone <repo-url>
   cd <repo-folder>
   ```
2. Install dependencies

   ```bash
   pip install -r requirements.txt
   ```
3. Run the Streamlit app

   ```bash
   streamlit run app.py
   ```

### 🔹 Contribution

This project can be extended by:

* Trying different ML models (Random Forest, XGBoost, Neural Networks)
* Hyperparameter tuning for better accuracy
* Enhancing Streamlit UI for improved user experience

---

✨ This PR adds the complete **Diabetes Prediction Logistic Regression project** with dataset preprocessing, model training, and deployment interface.

---

### CODE REVIEW

**Comment by MahraibFatima** [comment]
org GSSoC or SSoC?
